### PR TITLE
Addition of ThingSpeak Push to external push locations

### PR DESCRIPTION
**Pull request by johndoyle**
Adding the ability to push fermentrack temperature data from a brewpi device to a ThingSpeak channel.  I have only configured for the temp data to be sent at this point as I have no gravity sensor to test with.

### CODE REVIEW

File: docs/source/develop/push_support.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ Fermentrack currently supports three push targets:
 - **"Generic" Push Target** - Fermentrack's "native" push format - Pushes both specific gravity & temperature data
 - **Brewer's Friend** - Fermentrack's "native" push format - Pushes both specific gravity & temperature data
 - **Brewfather** - Fermentrack's "native" push format - Pushes both specific gravity & temperature data
+- **ThingSpeak** - Fermentrack's "native" push format - Pushes temperature data
```
Comment (thorrak) [inline docs/source/develop/push_support.rst]: Oh wow. I actually need to rewrite all of these. I think I just copied & pasted, when I meant to actually provide some kind of explanation of what each was. Sorry about that!

File: external_push/admin.py [hunk 2]
```diff
@@ -13,4 +13,8 @@ class BrewersFriendPushTargetAdmin(admin.ModelAdmin):
 
 @admin.register(BrewfatherPushTarget)
 class BrewfatherPushTargetAdmin(admin.ModelAdmin):
-    list_display = ('gravity_sensor_to_push', 'status', 'push_frequency')
\ No newline at end of file
+    list_display = ('gravity_sensor_to_push', 'status', 'push_frequency')
+    
+#@admin.register(ThingSpeakPushTarget)
```
Comment (thorrak) [inline external_push/admin.py]: Out of curiosity, why did you  comment out the admin?

File: external_push/models.py [hunk 3]
```diff
@@ -471,3 +471,160 @@ def send_data(self):
...
+
+    SENSOR_SELECT_CHOICES = (
+        (SENSOR_SELECT_ALL, "All Active Sensors/Devices"),
+        (SENSOR_SELECT_LIST, "Specific Sensors/Devices"),
+        (SENSOR_SELECT_NONE, "Nothing of this type"),
+    )
+
+    PUSH_FREQUENCY_CHOICES = (
+        # (30-1,    '30 seconds'),
+        (60 - 1, '1 minute'),
```
Comment (thorrak) [inline external_push/models.py]: Do you happen to know if ThingSpeak has a "minimum" push frequency? Is 1 minute too short, or is it allowed?

**Comment by thorrak** [comment]
Looks good! Thanks for getting this added! I'll get this merged in shortly.

File: external_push/models.py (hunk 3 above)
Comment (johndoyle) [reply external_push/models.py]: The free tier provides for 3,000,000 pushes per year  (8,219 messages per day) and up to 4 channels sharing that data.  I do not believe there is a push frequency limit published. (Read that I can't find it)

File: external_push/admin.py (hunk 2 above)
Comment (johndoyle) [reply external_push/admin.py]: I forgot to comment when debugging... Sorry - needed to update this line also
`    list_display = ('name', 'push_frequency', 'api_key', 'brewpi_to_push')`

File: external_push/models.py (hunk 3 above)
Comment (thorrak) [reply external_push/models.py]: 8219 would be roughly once every 30 seconds, so I think we’re good at 1/min!

Comment (johndoyle) [reply external_push/models.py]: The limit works for 1 minute but if going for multiple channels (Brewpis) the user would have to draw back on the update frequency if using all the time.  I did notice that you have to create a beer profile before the push works.  I had a lot of head scratching as I had my fridge on "Fridge Constant" without a beer profile.   Makes scene once I did create one.

**Comment by johndoyle** [comment]
Thanks.  Great project and I hope I can keep contributing.  Had to learn Python to do this so was fun :-)

**Comment by thorrak** [comment]
This has been merged into a new branch - new_pushtargets. I'm going to merge in another (slightly conflicting) pull request for Grainfather support, then test both this coming weekend before pushing to dev

**Pull request merged**